src/memio.c

```diff
@@ -182,7 +182,7 @@ int mem_fseek(MEMFILE *stream, signed long position, mem_rel_t whence)
 			return -1;
 	}
 
-	if (newpos < stream->buflen)
+	if (newpos <= stream->buflen)
```

Comment:
This appears to be a typo and has already been fixed in other sources ports (Woof, Odamex). Now the behavior matches standard C behavior (e.g. `fseek()`). Other than this PR, `mem_fseek()` is not used anywhere else in the code base.